typing/typecore.ml

```diff
@@ -5217,19 +5560,92 @@ and type_cases
       check_unused ~lev allow_modules env ty_arg_check val_cases ;
       check_unused ~lev allow_modules env Predef.type_exn exn_cases ;
     end;
-    Parmatch.check_ambiguous_bindings val_cases ;
```

Comment:
These checks are no longer performed on `fun (Some pattern) -> ...`, I think. Is that OK?

Reply:
OK. Please resolve.